syntax = "proto3";

package fl_client;

service ClientService {
  // 1. Receive global model from server
  rpc ReceiveGlobalModel(ReceiveGlobalModelRequest) returns (ReceiveGlobalModelResponse);

  // 2. Get trained model from client
  rpc ReceiveModelForAccuracyBasedMeasure(AccuracyBasedMeasureRequest) returns (AccuracyBasedMeasureResponse);
}

message ReceiveGlobalModelRequest {
  int32 client_id = 1;
  bytes global_model = 2;
  string model_type = 3;
}

message ReceiveGlobalModelResponse {
    int32 client_id = 1;
    bytes client_model = 2;
    float training_time = 3;
    float iid_measure = 4;
}

message AccuracyBasedMeasureRequest {
  int32 client_id = 1;
  bytes model = 2;
  string model_type = 3;
}

message AccuracyBasedMeasureResponse {
  float weighted_val_acc = 1;
}